modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1341,12 +1426,12 @@ void ONNXImporter::populateNet(Net dstNet)
 
                 opencv_onnx::NodeProto proto;
                 proto.add_output(constParams.name);
-                addLayer(dstNet, constParams, proto, layer_id, outShapes);
+                addLayer(constParams, proto);
 
                 layerParams.type = "Scale";
                 layerParams.set("bias_term", false);
                 node_proto.set_input(0, constParams.name);
-                node_proto.set_input(1, shapeIt->first);
+                node_proto.set_input(1, srcName);
```

Comment:
@l-bat Please take a look on this change.
(`shapeIt` scope was too large, its locality is reduced to avoid errors and dangling pointers)